plugin/advisor/pg/advisor_table_no_fk.go

```diff
@@ -0,0 +1,108 @@
+package pg
+
+import (
+	"fmt"
+
+	"github.com/bytebase/bytebase/plugin/advisor"
+	"github.com/bytebase/bytebase/plugin/advisor/db"
+	"github.com/bytebase/bytebase/plugin/parser/ast"
+)
+
+var (
+	_ advisor.Advisor = (*TableNoFKAdvisor)(nil)
+	_ ast.Visitor     = (*tableNoFKChecker)(nil)
+)
+
+func init() {
+	advisor.Register(db.Postgres, advisor.PostgreSQLTableNoFK, &TableNoFKAdvisor{})
+}
+
+// TableNoFKAdvisor is the advisor checking table disallow foreign key.
+type TableNoFKAdvisor struct {
+}
+
+// Check checks table disallow foreign key.
+func (*TableNoFKAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	stmts, errAdvice := parseStatement(statement)
+	if errAdvice != nil {
+		return errAdvice, nil
+	}
+
+	level, err := advisor.NewStatusBySQLReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+
+	checker := &tableNoFKChecker{
+		level: level,
+		title: string(ctx.Rule.Type),
+	}
+
+	for _, stmt := range stmts {
+		checker.text = stmt.Text()
+		ast.Walk(checker, stmt)
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    advisor.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+	return checker.adviceList, nil
+}
+
+type tableNoFKChecker struct {
+	adviceList []advisor.Advice
+	level      advisor.Status
+	title      string
+	text       string
+}
+
+// Visit implements the ast.Visitor interface.
+func (checker *tableNoFKChecker) Visit(node ast.Node) ast.Visitor {
+	var hasFK *ast.TableDef
```

Comment:
How about use another name for `hasFK` variable? It looks like a bool instead of the `TableDef`